library/src/main/java/com/airbnb/android/airmapview/RuntimePermissionUtils.java

```diff
@@ -0,0 +1,99 @@
+package com.airbnb.android.airmapview;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.Fragment;
+
+import static android.support.v4.content.PermissionChecker.checkSelfPermission;
+
+/**
+ * Utility class that handles Runtime location permissions for Android M
+ */
+public final class RuntimePermissionUtils {
+
+  private RuntimePermissionUtils() {
+  }
+
+  private static final byte LOCATION_PERMISSION_REQUEST_CODE = 1;
+  private static final String[] LOCATION_PERMISSIONS =
+          new String[]{"android.permission.ACCESS_FINE_LOCATION", "android.permission.ACCESS_COARSE_LOCATION"};
+
+  /**
+   * Checks all given permissions have been granted.
+   *
+   * @param grantResults results
+   * @return returns true if all permissions have been granted.
+   */
+  public static boolean verifyPermissions(int... grantResults) {
+    for (int result : grantResults) {
+      if (result != PackageManager.PERMISSION_GRANTED) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns true if the Activity or Fragment has access to all given permissions.
+   *
+   * @param context     context
+   * @param permissions permission list
+   * @return returns true if the Activity or Fragment has access to all given permissions.
+   */
+  public static boolean hasSelfPermissions(Context context, String... permissions) {
+    for (String permission : permissions) {
+      if (checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Checks given permissions are needed to show rationale.
+   *
+   * @param activity    activity
+   * @param permissions permission list
+   * @return returns true if one of the permission is needed to show rationale.
+   */
+  public static boolean shouldShowRequestPermissionRationale(Activity activity, String... permissions) {
+    for (String permission : permissions) {
+      if (ActivityCompat.shouldShowRequestPermissionRationale(activity, permission)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Call this method to execute code that needs locations permission, and put the executed code in
+   * onLocationPermissionsGranted().
+   */
+  public static void checkLocationPermissions(Fragment target, AirMapInterface airMapInterface) {
```

Comment:
pass in a `Context` here instead of a `Fragment`, since that is what really is needed and makes it possible to use this method from an `Activity`